api/v1alpha1/etcdpeer_types.go

```diff
@@ -81,6 +94,25 @@ type EtcdPeerList struct {
 	Items           []EtcdPeer `json:"items"`
 }
 
+var _ webhook.Defaulter = &EtcdPeer{}
+
+// Default sets default values for optional EtcdPeer fields.
+// This is used in webhooks and in the Reconciler to ensure that nil pointers
+// have been replaced with concrete pointers.
+// This avoids nil pointer panics later on.
+func (o *EtcdPeer) Default() {
+	if o.Spec.Storage.VolumeClaimTemplate.AccessModes == nil {
```

Comment:
worth a nil check on o.Spec.Storage too?

Reply:
Done, but I've decided to add Validation which will reject any EtcdPeer with a missing required field, before we attempt to default it.